pytorch_lightning/metrics/metric.py

```diff
@@ -238,17 +245,26 @@ def compute(self):  # pylint: disable=E0202
         """
         pass
 
-    def reset(self):
-        """
-        This method automatically resets the metric state variables to their default value.
-        """
+    def _reset_to_default(self):
+        """ reset metric state to their default values """
         for attr, default in self._defaults.items():
             current_val = getattr(self, attr)
             if isinstance(current_val, torch.Tensor):
                 setattr(self, attr, deepcopy(default).to(current_val.device))
             else:
                 setattr(self, attr, deepcopy(default))
 
+    def reset(self):
+        """
+        This method automatically resets the metric state variables to their default value.
+        """
+        if self.auto_reset_on_compute or not self._internal_reset:
+            reset_fn = self._reset_to_default
+        else:
+            reset_fn = all_except_rank_zero(self._reset_to_default)
```

Comment:
Does this make it not reset on rank 0? Why would we not want to reset on rank 0?